sentry_sdk/utils.py

```diff
@@ -1572,6 +1573,56 @@ def parse_version(version):
     return release_tuple
 
 
+def _generate_installed_modules():
+    # type: () -> Iterator[Tuple[str, str]]
+    try:
+        from importlib import metadata
+
+        for dist in metadata.distributions():
+            name = dist.metadata["Name"]
+            # `metadata` values may be `None`, see:
+            # https://github.com/python/cpython/issues/91216
+            # and
+            # https://github.com/python/importlib_metadata/issues/371
+            if name is not None:
+                version = metadata.version(name)
+                if version is not None:
+                    yield _normalize_module_name(name), version
+
+    except ImportError:
+        # < py3.8
+        try:
+            import pkg_resources
+        except ImportError:
+            return
+
+        for info in pkg_resources.working_set:
+            yield _normalize_module_name(info.key), info.version
+
+
+def _normalize_module_name(name):
+    # type: (str) -> str
+    return name.lower()
+
+
+def _get_installed_modules():
+    # type: () -> Dict[str, str]
+    global _installed_modules
+    if _installed_modules is None:
+        _installed_modules = dict(_generate_installed_modules())
+    return _installed_modules
```

Comment:
Would it perhaps be a good idea to place this code, along with the `_installed_modules` into a static class, so we can avoid having a global variable? I guess the static class would still effectively be storing a global state, so perhaps it makes only a small difference here

Reply:
We could wrap this in a static class but then it'd be a bit inconsistent to have some installed modules related code in the class (the `_get_installed_modules` function above + the `_installed_modules` var) and some of it outside (e.g. the `package_version` helper). We _could_ put everything in the class, but it's not great API to have to then import the class to use a small utility function.

In general there's good usecases for having a global var and this is IMO one of them -- I'd prefer to keep this as is if that's ok with you.